crates/sui-graphql-rpc/schema/current_progress_schema.graphql

```diff
@@ -266,6 +280,24 @@ type DynamicFieldEdge {
 	cursor: String!
 }
 
+input DynamicFieldName {
+	"""
+	A string flag of 'DynamicField' or 'DynamicObject'.
+	This is needed to disambiguate the child object,
+	as it is possible for a dynamic field and a dynamic object field to share the same name.
+	"""
+	kind: String!
```

Comment:
I'm not a big fan of this approach because we're using a string type to encode 1 bit of information -- it's just asking for someone to introduce a bug with a typo, and get a confusing error message, while if we exposed two separate functions, (`dynamicField` and `dynamicObjectField`) we get to take advantage of auto-complete in the editor, and error messages that spot typos and suggest alternatives:


```
{
  "data": null,
  "errors": [
    {
      "message": "Unknown field \"chainIdentifir\" on type \"Query\". Did you mean \"chainIdentifier\"?",
      "locations": [
        {
          "line": 2,
          "column": 2
        }
      ]
    }
  ]
}
```